homeassistant/components/unifi/device_tracker.py

```diff
@@ -66,10 +67,65 @@
     EventKey.WIRELESS_GUEST_ROAM,
     EventKey.WIRELESS_GUEST_ROAMRADIO,
 )
+WIRELESS_DISCONNECTION = (
+    EventKey.WIRELESS_CLIENT_DISCONNECTED,
+    EventKey.WIRELESS_GUEST_DISCONNECTED,
+)
+
+
+@callback
+def async_client_allowed_fn(controller: UniFiController, obj_id: str) -> bool:
+    """Check if client is allowed."""
+    client = controller.api.clients[obj_id]
+    if not controller.option_track_clients:
+        return False
+
+    if client.mac not in controller.wireless_clients:
+        if not controller.option_track_wired_clients:
+            return False
 
+    elif (
+        client.essid
+        and controller.option_ssid_filter
+        and client.essid not in controller.option_ssid_filter
+    ):
+        return False
 
-_DataT = TypeVar("_DataT", bound=Device)
-_HandlerT = TypeVar("_HandlerT", bound=Devices)
+    return True
+
+
+@callback
+def async_client_heartbeat_timedelta_fn(
+    controller: UniFiController, obj_id: str
+) -> timedelta:
+    """Check if device object is disabled."""
+    return controller.option_detection_time
+
+
+@callback
+def async_client_is_connected_fn(controller: UniFiController, obj_id: str) -> bool:
+    """Check if device object is disabled."""
+    client = controller.api.clients[obj_id]
+
+    if client.is_wired != (is_wired := client.mac not in controller.wireless_clients):
+        if not controller.option_ignore_wired_bug:
+            return False  # Wired bug in action
+        is_wired = client.is_wired
```

Comment:
This was a bit hard to reason about on the first pass. Moving the assignment might help: 

```python
if (is_wired := client.mac not in controller.wireless_clients) != client.is_wired:
```

Reply:
I think I prefer it as it is with the primary value on the left